### PR TITLE
Fix GNT lock in FundsLocker

### PR DESCRIPTION
**Pull request by Krigpl**
Current GNT lock value depends on the number of the unfinished tasks which is being updated but not taken into calculations after #2774. This PR fixes that.

### CODE REVIEW

File: golem/transactions/ethereum/fundslocker.py [hunk 1]
```diff
@@ -11,10 +11,14 @@
 
 class TaskFundsLock:  # pylint: disable=too-few-public-methods
     def __init__(self, task):
-        self.gnt_lock = task.price
+        self.price = task.subtask_price
```
Comment (maaktweluit) [inline golem/transactions/ethereum/fundslocker.py:14]: is the name `price` here clear enough or would `subtask_price` be better?

File: tests/golem/transactions/ethereum/test_fundslocker.py [hunk 2]
```diff
@@ -28,15 +28,15 @@ def test_lock_funds(self):
...
+        task.subtask_price = 320
         task.total_tasks = 10
         task.header.deadline = task_deadline
         fl.lock_funds(task)
         tfl = fl.task_lock['abc']
 
         def test_params(tfl):
             assert isinstance(tfl, TaskFundsLock)
-            assert tfl.gnt_lock == 320
+            assert tfl.gnt_lock == 320 * 10
```
Comment (maaktweluit) [inline tests/golem/transactions/ethereum/test_fundslocker.py:39]: maybe the magic numbers can be replaced with `task.*`?

**Review by maaktweluit — COMMENTED** [review]
LGTM!

2 small comments, will approve on reply

File: golem/transactions/ethereum/fundslocker.py (hunk 1 above)
Comment (Krigpl) [reply golem/transactions/ethereum/fundslocker.py:14]: I dunno, it was `price` before, either is fine. But also I'm aiming to change the way FundsLocker works altogether (as you might have noticed already) so this should't matter in the long run

File: tests/golem/transactions/ethereum/test_fundslocker.py (hunk 2 above)
Comment (Krigpl) [reply tests/golem/transactions/ethereum/test_fundslocker.py:39]: Yeah, but as I mentioned when I'm finished with FundsLocker most of this stuff won't exist anymore so I didn't bother to fix it now :)

**Review by maaktweluit — APPROVED** [review]
Thank you for the replies, looks good to me!

**Review by jiivan — APPROVED** [review]

**Pull request merged**